dali/base/sysinfologger.cpp

```diff
@@ -0,0 +1,682 @@
+    {
+        return haveDateRange;
+    }
+    virtual bool isInDateRange(unsigned __int64 ts) const override
+    {
+        unsigned tyear, tmonth, tday;
+        extractDate(ts, tyear, tmonth, tday);
+
+        if (matchStartYear)
+        {
+            if (tyear<matchStartYear)
+                return false;
+            if (tyear==matchStartYear)
+            {
+                if (matchStartMonth)
+                {
+                    if (tmonth<matchStartMonth)
+                        return false;
+                    if (tmonth==matchStartMonth)
+                    {
+                        if (matchStartDay && tday<matchStartDay)
+                            return false;
+                    }
+                }
+            }
+        }
+        if (matchEndYear)
+        {
+            if (tyear>matchEndYear)
+                return false;
+            if (tyear==matchEndYear)
+            {
+                if (matchEndMonth)
+                {
+                    if (tmonth>matchEndMonth)
+                        return false;
+                    if (tmonth==matchEndMonth)
+                    {
+                        if (matchEndDay && tday>matchEndDay)
+                            return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+    virtual bool queryHiddenOnly() const override
+    {
+        return hiddenOnly;
+    }
+    virtual bool queryVisibleOnly() const override
+    {
+        return visibleOnly;
+    }
+    virtual unsigned __int64 queryMatchTimeStamp() const override
+    {
+        return matchTimeStamp;
+    }
+    virtual unsigned queryStartYear() const override
+    {
+        return matchStartYear;
+    }
+    virtual unsigned queryStartMonth() const override
+    {
+        return matchStartMonth;
+    }
+    virtual unsigned queryStartDay() const override
+    {
+        return matchStartDay;
+    }
+    virtual unsigned queryEndYear() const override
+    {
+        return matchEndYear;
+    }
+    virtual unsigned queryEndMonth() const override
+    {
+        return matchEndMonth;
+    }
+    virtual unsigned queryEndDay() const override
+    {
+        return matchEndDay;
+    }
+    virtual const char * queryMatchSource() const override
+    {
+        return matchSource.str();
+    }
+    virtual LogMsgCode queryMatchCode() const override
+    {
+        return matchCode;
+    }
+    virtual LogMsgAudience queryMatchAudience() const override
+    {
+        return matchAudience;
+    }
+    virtual LogMsgClass queryMatchClass() const override
+    {
+        return matchClass;
+    }
+    virtual StringBuffer & getQualifierXPathFilter(StringBuffer & xpath) const override
+    {
+        bool fullDayMatch=false;
+        bool hardMatchYear = matchStartYear && (matchStartYear==matchEndYear);
+        bool hardMatchMonth = matchStartMonth && (matchStartMonth==matchEndMonth);
+        if (hardMatchYear && hardMatchMonth)
```

Comment:
@jakesmith is there any scope for m%04u* if the month is not known?

Reply:
yes, that would make sense, it would still fetch all at the client level (via lazy fetch), but at the iterator wouldn't see any non-matching years, so would improve the efficiency of this code.

But also, it would probably be worth using iptiter_remote (or probably iptiter_remotegetbranch) to getElements. That will cause the filtering to be done at the server side, which means, if there's many parent nodes that don't have any results below them, it will avoid the client pulling them down as it iterates.